internal/gitrepo/gitrepo.go

```diff
@@ -479,3 +481,33 @@ func (r *LocalRepository) Restore(paths []string) error {
 	cmd.Dir = r.Dir
 	return cmd.Run()
 }
+
+// CleanUntracked removes untracked files within the given paths.
+func (r *LocalRepository) CleanUntracked(paths []string) error {
+	slog.Info("Cleaning untracked files", "paths", strings.Join(paths, ","))
+	worktree, err := r.repo.Worktree()
+	if err != nil {
+		return err
+	}
+
+	statue, err := worktree.Status()
+	if err != nil {
+		return err
+	}
+
+	// Remove an untracked file if it lives within one of the given paths.
+	for _, path := range paths {
+		for file, fileStatue := range statue {
+			if !strings.Contains(file, path) || fileStatue.Worktree != git.Untracked {
+				continue
+			}
+
+			relPath := filepath.Join(r.Dir, file)
+			if err := os.Remove(relPath); err != nil {
+				return fmt.Errorf("failed to remove untracked file, %s: %q", relPath, err)
```

Comment:
qq, what makes a file untracked in this case? Would it make sense to potentially leave an untracked file and proceed the rest of the flow?

Reply:
Ok gotcha, I think I understand. Thanks for the clarification!

Last question: Does build run after all the libraries are generated? Or does it run after a single library is generated? Since I recall we do allow failures during generation, would erroring out when cleaning a single library prevent other libraries from generating? If so, is that intended?

Changes LGTM, asking questions for my curiosity. If this is intended, resolve this comment and merge :)